### PR TITLE
Visualization

### PR DESCRIPTION
**Pull request by nateslo**
Added "atom" animation and progress bar. On "complete", screen flash and nodes converge. Will begin working on "packets" animation next

### CODE REVIEW

File: client/scripts/progress.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+var colors = {
```
Comment (ekong2) [inline client/scripts/progress.js]: Is it possible to add inline comments to blocks just giving general description of what they are?

File: client/scripts/visualization.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
+var width = window.innerWidth,
```
Comment (ekong2) [inline client/scripts/visualization.js]: same with visualization.js, is it possible to add inline comments at the beginning of every block explaining what the code does or what it's for?

**Comment by ekong2** [comment]
i love love love the animations and the glow. too awesome
we won't have a check in tomorrow so I'll be leaving my thoughts here.
two last requests:
1. can you make the screen flash upon receiving the 'data' event? (make ppls spazz and admire your handiwork in awe) also on a more serious note, to break the monotony if the entire operation from 0-100% takes a long time and for people to have a more concrete feeling of "oh! i received something and i'm contributing now". i wasn't too sure but did you implement the 'data' event in the form of the balls suddenly going more crazy??
2. can you make the background turn white or a cool ionic blue after it all ends? (just to give it a sense of completeness)

I feel more strongly about #2 than I do about #1 (also because #2 is simpler) but in the end, you're the designer and you get the final say in the design choices for the front end. If you feel #1 might make it too spazzstic if the screen flashes every 3 seconds then lets leave it be. If you don't like #2, that's okay too.

Bottom line, you're the man Nate. You decide :)

File: client/scripts/slvProc.js [hunk 3]
```diff
@@ -51,14 +53,20 @@ socket.on('data', function(data) {
...
-  console.log("Clients: ",connectedClients)
+  console.log("Clients: ",connectedClients);
+  updateConnected(connectedClients);
 });
 
 socket.on('complete', function(){
+  var btn = document.getElementById("rdy");
+  btn.innerHTML = 'Complete';
   console.log("COMPLETE");
+  stopAnim();
```
Comment (andyttran) [inline client/scripts/slvProc.js]: We will have to modularize the slave process as well. Thus keeping the files separate and importing them in is the optimal approach which is being done here. We may just need to take the start and stop animations out of the core slave module.

File: client/scripts/visualization.js (hunk 2 above)
Comment (nateslo) [inline client/scripts/visualization.js]: Absolutely, I agree these files need some cleaning up

File: client/scripts/slvProc.js (hunk 3 above)
Comment (nateslo) [inline client/scripts/slvProc.js]: Yeah, might be a good time to implement a little mvc to clean these files up. I'll work on that, as well as commenting.
